### ISSUE TITLE
Can we store string representation of event?

### ISSUE THREAD

**Issue by faizanakram99** [monteiro/ddd-bundle#2]
https://github.com/monteiro/ddd-bundle/blob/d13ad0b213fd77b80aa99a815cdb5d5a5b0babf4/src/Infrastructure/Console/PublishDomainEventsCommand.php#L79

Can we add the possibility of storing type as simple strings and map them to real class somewhere else? (Of course the bundle needs to be made aware of all domain events in that case)

Maybe add a static method/property (`getName` / `eventName`) method to DomainEvent?

Nice to have but not necessary, helps when events are moved around (say namespace change) and say you want to create a projection from the stored events at some point.

**Comment by monteiro** [comment]
> Can we add the possibility of storing type as simple strings and map them to real class somewhere else? (Of course the bundle needs to be made aware of all domain events in that case)

Normally we save as json, then you can map as you wish. The goal of the event is to be very simple, just passing data, not huge complex objects.


> Maybe add a static method/property (getName / eventName) method to DomainEvent?

It's the `$storedEvent->getTypeName()` the name of the event. Maybe I should change the method name to be more clear. getEventName()

**Comment by faizanakram99** [comment]
> > Can we add the possibility of storing type as simple strings and map them to real class somewhere else? (Of course the bundle needs to be made aware of all domain events in that case)
> 
> Normally we save as json, then you can map as you wish. The goal of the event is to be very simple, just passing data, not huge complex objects.
> 
> > Maybe add a static method/property (getName / eventName) method to DomainEvent?
> 
> It's the `$storedEvent->getTypeName()` the name of the event. Maybe I should change the method name to be more clear. getEventName()

yes it is fully qualified class name of the event class, right? By event name I meant more of a natural representation of the event ... human readable (like `order.created`, `invoice.sent`, etc), so that even if we change the class name of event or move it to a different namespace, we wouldn't need to update existing db records from `event_store` table.

It is not too important but would be nice

**Comment by monteiro** [comment]
the challenge of using order.created and invoice.sent events is that you will need some kind of mapping, which get_type($class) is not needed. How would you do it? I want this to be as less config as possible. Maybe in the future we can add that possibility, or now if you give me a good idea how to implement it, without requiring manual mapping.

Changed typeName to be eventName to be more specific: https://github.com/monteiro/ddd-bundle/commit/b84798247ca72bf2492b17c7ffb37db2746d0d74 following your suggestion.
